Emby.Server.Implementations/AppBase/BaseConfigurationManager.cs

```diff
@@ -98,16 +98,31 @@ protected BaseConfigurationManager(IApplicationPaths applicationPaths, ILoggerFa
         public IApplicationPaths CommonApplicationPaths { get; private set; }
 
         /// <summary>
-        /// Gets the system configuration
+        /// Gets the system configuration.
         /// </summary>
         /// <value>The configuration.</value>
         public BaseApplicationConfiguration CommonConfiguration
         {
             get
             {
-                // Lazy load
-                LazyInitializer.EnsureInitialized(ref _configuration, ref _configurationLoaded, ref _configurationSyncLock, () => (BaseApplicationConfiguration)ConfigurationHelper.GetXmlConfiguration(ConfigurationType, CommonApplicationPaths.SystemConfigurationFilePath, XmlSerializer));
-                return _configuration;
+                if (_configurationLoaded)
+                {
+                    return _configuration;
+                }
+
+                lock (_configurationSyncLock)
+                {
+                    if (_configurationLoaded)
+                    {
+                        return _configuration;
+                    }
+
+                    _configuration = (BaseApplicationConfiguration)ConfigurationHelper.GetXmlConfiguration(ConfigurationType, CommonApplicationPaths.SystemConfigurationFilePath, XmlSerializer);
```

Comment:
are you telling me C# doesn't have some well-defined Singleton support, so you had to hand-roll it?..

Reply:
? This just ensures the loading of the config is thread safe.

Reply:
Okay, I checked and C# has a few ways of making a singleton but nothing completely built-in or fitting, let's close this conversation on that.